### PR TITLE
Improve TLDEventDataFilter's query field parse

### PR DESCRIPTION
**Pull request by apmoriarty**
This addresses a defect where the `termOffsetMap` portion of a content function was considered a query field
- 5a7b4f4351f5ed3d9fb3c07124426215c4429869 correct defect 
- d7f2479255797cd1eb5a57cedc19414ceeb024b5 update test to address sonar/intellij warnings

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/query/predicate/TLDEventDataFilter.java [hunk 1]
```diff
@@ -618,6 +618,23 @@ private void extractQueryFieldsFromScript(ASTJexlScript script) {
...
     
+    /**
+     * Fields are always uppercase
+     * 
+     * @param field
+     *            the field
+     * @return true if all characters are uppercase
+     */
+    private boolean containsLowers(String field) {
+        for (char c : field.toCharArray()) {
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/predicate/TLDEventDataFilter.java]: This is hacky.  Lets use that QueryFieldVisitor that we created a year ago....

**Review by ivakegg — CHANGES_REQUESTED** [review]

File: warehouse/query-core/src/test/java/datawave/query/predicate/TLDEventDataFilterTest.java [hunk 2]
```diff
@@ -594,4 +590,18 @@ public void apply_acceptGroupingFields() throws ParseException {
         assertFalse(result3);
         assertTrue(result4);
     }
+    
+    @Test
+    public void testBadIdentifier() throws ParseException {
```
Comment (avgAGB) [inline warehouse/query-core/src/test/java/datawave/query/predicate/TLDEventDataFilterTest.java:603]: Would it make sense to add a similar test to this with an identifier that had mixed capitalization in it, to confirm that those fields would be rejected as well?

File: warehouse/query-core/src/main/java/datawave/query/predicate/TLDEventDataFilter.java (hunk 1 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/predicate/TLDEventDataFilter.java]: Opting to use the configured fields in the QueryIterator per our discussion

File: warehouse/query-core/src/test/java/datawave/query/predicate/TLDEventDataFilterTest.java (hunk 2 above)
Comment (apmoriarty) [reply warehouse/query-core/src/test/java/datawave/query/predicate/TLDEventDataFilterTest.java:603]: Can do

**Review by avgAGB — APPROVED** [review]

**Review by ivakegg — APPROVED** [review]

**Pull request merged**